src/huggingface_hub/hub_mixin.py

```diff
@@ -22,26 +22,9 @@
 class ModelHubMixin:
     def __init__(self, *args, **kwargs):
         """
-        Mix this class with your torch-model class for ease process of saving & loading from huggingface-hub
-
-        Example::
-
-            >>> from huggingface_hub import ModelHubMixin
-
-            >>> class MyModel(nn.Module, ModelHubMixin):
-            ...    def __init__(self, **kwargs):
-            ...        super().__init__()
-            ...        self.config = kwargs.pop("config", None)
-            ...        self.layer = ...
-            ...    def forward(self, ...)
-            ...        return ...
-
-            >>> model = MyModel()
-            >>> model.save_pretrained("mymodel", push_to_hub=False) # Saving model weights in the directory
-            >>> model.push_to_hub("mymodel", "model-1") # Pushing model-weights to hf-hub
-
-            >>> # Downloading weights from hf-hub & model will be initialized from those weights
-            >>> model = MyModel.from_pretrained("username/mymodel@main")
+        A Generic Base Model Hub Mixin. Define your own mixin for anything by inheriting from this class
+        and overwriting _from_pretrained and _save_pretrained to define custom logic for saving/loading
+        your classes. See ``huggingface_hub.PyTorchModelHubMixin`` for an example.
```

Comment:
Unrelated to your PR: should this comment live in the `__init__` as is the case here, or should it be the class-wide documentation? I'd favor the latter as this docstring isn't linked to the init

Reply:
Yea lets go with class-wide. I think we can actually get rid of `__init__` entirely. It causes problems for some integrations (sklearn for example, which inspects the init args)